src/bson_util/mod.rs

```diff
@@ -61,12 +63,15 @@ pub(crate) fn update_document_check(update: &Document) -> Result<()> {
     }
 }
 
-pub(crate) fn serialize_duration_as_i64_millis<S: Serializer>(
+pub(crate) fn serialize_duration_as_int_millis<S: Serializer>(
     val: &Option<Duration>,
     serializer: S,
 ) -> std::result::Result<S::Ok, S::Error> {
     match val {
-        Some(duration) => serializer.serialize_i64(duration.as_millis() as i64),
+        Some(duration) if duration.as_millis() > i32::MAX as u128 => {
```

Comment:
Since our extjson parser now parses integers into the smallest width possible, I needed to update how we serialized durations for the purpose of tests. This shouldn't have any external impact on users, as the server is happy to read a timeout option as a 32-bit integer instead of a 64-bit one.

There are dozens of similar changes below; I didn't bother calling them all out.